.github/workflows/chekc-fork-branch.yaml

```diff
@@ -0,0 +1,22 @@
+name: Check Fork Branch
+
+on:
+  pull_request:
+    types: [ opened, synchronize, reopened ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Echo Branch Configuration
+        id: branch_configuration
+        run: |
+          echo "BRANCH_NAME: ${{github.head_ref}}"
+          echo "REPO: ${{github.event.pull_request.head.repo.full_name}}"
+          echo "BASE_REF: ${{github.event.pull_request.base.ref}}"
+      - name: Check fork branch
+        if: github.event.pull_request.head.repo.full_name != github.repository && github.head_ref == 'main'
+        uses: actions/github-script@v3
+        with:
+          script: |
+            core.setFailed('main branch should not be used in pull requests from a fork"')
```

Comment:
Nice. I hadn't seen core used before.  I needed to look it up.  In case anyone else is in the same boat....

https://github.com/actions/toolkit/tree/main/packages/core#exit-codes